events/janus_gelfevh.c

```diff
@@ -0,0 +1,590 @@
+	janus_config *config = janus_config_parse(filename);
+	if(config == NULL) {
+		JANUS_LOG(LOG_WARN, "Couldn't find .jcfg configuration file (%s), trying .cfg\n", JANUS_GELFEVH_PACKAGE);
+		g_snprintf(filename, 255, "%s/%s.cfg", config_path, JANUS_GELFEVH_PACKAGE);
+		JANUS_LOG(LOG_VERB, "Configuration file: %s\n", filename);
+		config = janus_config_parse(filename);
+	}
+	if(config != NULL) {
+		/* Handle configuration */
+		janus_config_print(config);
+		janus_config_category *config_general = janus_config_get_create(config, NULL, janus_config_type_category, "general");
+
+		/* Setup the sample event handler, if required */
+		janus_config_item *item;
+		item = janus_config_get(config, config_general, janus_config_type_item, "enabled");
+		if(!item || !item->value || !janus_is_true(item->value)) {
+			JANUS_LOG(LOG_WARN, "GELF event handler disabled (Janus API)\n");
+			goto done;
+		}
+		item = janus_config_get(config, config_general, janus_config_type_item, "backend");
+		if(!item || !item->value) {
+			JANUS_LOG(LOG_WARN, "Missing or invalid backend\n");
+			goto done;
+		}
+		backend = g_strdup(item->value);
+		item = janus_config_get(config, config_general, janus_config_type_item, "port");
+		if(!item || !item->value) {
+			JANUS_LOG(LOG_WARN, "Missing or invalid port\n");
+			goto done;
+		}
+		port = g_strdup(item->value);
+		item = janus_config_get(config, config_general, janus_config_type_item, "protocol");
+		if (item && item->value) {
+			if (strcasecmp(item->value, "udp") == 0){
+				transport = JANUS_GELFEVH_SOCKET_TYPE_UDP;
+			} else if (strcasecmp(item->value, "tcp") == 0){
+				transport = JANUS_GELFEVH_SOCKET_TYPE_TCP;
+			} else {
+				JANUS_LOG(LOG_WARN, "Missing or invalid transport, using default:UDP\n");
+			}
+		}
+		item = janus_config_get(config, config_general, janus_config_type_item, "max_message_len");
+		if (item && item->value) {
+			if(atoi(item->value) == 0) {
+				JANUS_LOG(LOG_WARN, "Missing or invalid max_message_len, using default: %d\n", max_gelf_msg_len);
+			} else {
+				max_gelf_msg_len = atoi(item->value);
+			}
+		}
+		/* Which events should we subscribe to? */
+		item = janus_config_get(config, config_general, janus_config_type_item, "events");
+		if(item && item->value)
+			janus_events_edit_events_mask(item->value, &janus_gelfevh.events_mask);
+		/* Compact, so no spaces between separators */
+		json_format = JSON_COMPACT | JSON_PRESERVE_ORDER;
+
+		/* Check if we need any compression */
+		item = janus_config_get(config, config_general, janus_config_type_item, "compress");
+		if(item && item->value && janus_is_true(item->value)) {
+			compress = TRUE;
+			item = janus_config_get(config, config_general, janus_config_type_item, "compression");
+			if(item && item->value) {
+				int c = atoi(item->value);
+				if(c < 0 || c > 9) {
+					JANUS_LOG(LOG_WARN, "Invalid compression factor '%d', falling back to '%d'...\n", c, compression);
+				} else {
+					compression = c;
+				}
+			}
```

Comment:
This should have a check for TCP, and show a warning that, if TCP is used and compression enabled, data will actually not be compressed (as explained in the comment at the beginning of the code). People unaware of the limitation may expect the data to be compressed while it won't be, so a warning will clarify the actual behaviour.